src/modules/commander/commander.cpp

```diff
@@ -365,98 +385,51 @@ int commander_main(int argc, char *argv[])
 
 	if (!strcmp(argv[1], "arm")) {
 		if (TRANSITION_CHANGED != arm_disarm(true, &mavlink_log_pub, "command line")) {
-			warnx("arming failed");
+			PX4_WARN("arming failed");
 		}
 		return 0;
 	}
 
 	if (!strcmp(argv[1], "disarm")) {
 		if (TRANSITION_DENIED == arm_disarm(false, &mavlink_log_pub, "command line")) {
-			warnx("rejected disarm");
+			PX4_WARN("rejected disarm");
 		}
 		return 0;
 	}
 
 	if (!strcmp(argv[1], "takeoff")) {
 
+		bool ret = false;
+
 		/* see if we got a home position */
 		if (status_flags.condition_local_position_valid) {
 
 			if (TRANSITION_DENIED != arm_disarm(true, &mavlink_log_pub, "command line")) {
-
-				struct vehicle_command_s cmd = {
-					.timestamp = hrt_absolute_time(),
-					.param5 = (double)NAN,
-					.param6 = (double)NAN,
-					/* minimum pitch */
-					.param1 = NAN,
-					.param2 = NAN,
-					.param3 = NAN,
-					.param4 = NAN,
-					.param7 = NAN,
-					.command = vehicle_command_s::VEHICLE_CMD_NAV_TAKEOFF,
-					.target_system = status.system_id,
-					.target_component = status.component_id
-				};
-
-				orb_advert_t h = orb_advertise_queue(ORB_ID(vehicle_command), &cmd, vehicle_command_s::ORB_QUEUE_LENGTH);
-				(void)orb_unadvertise(h);
```

Comment:
This is missing now

Reply:
That was intentional, unadvertise doesn't seem to do anything other than mark it unpublished.

Reply:
Yes, exactly. In future it might do more, but since the `vehicle_command` is a bit special, it's ok not to use it here.

Reply:
Isn't that potentially problematic though? If you advertise/publish a vehicle_command then immediately unadvertise another module could conceivably miss the command (depending on actual implementation and other vehicle_command usage in the system).

Reply:
I think I misinterpreted your initial comment. Are you suggesting unadvertising everywhere, or never unadvertising for vehicle_command?

Reply:
> Are you suggesting unadvertising everywhere, or never unadvertising for vehicle_command?

Never for vehicle_command.

Reply:
Agreed.